llvm/lib/Transforms/IPO/GlobalOpt.cpp

```diff
@@ -1901,6 +1902,20 @@ static void makeAllConstantUsesInstructions(Constant *C) {
   }
 }
 
+static bool IsRepo = false;
+
+/// Tests whether we're producing output for a Program Repository.
+static bool isObjFormatRepo(Module &M) {
+  return Triple(M.getTargetTriple()).isOSBinFormatRepo();
+}
+
+/// Reset the global varible GV TicketNode metadata.
+static void calculateGlobalDigest(GlobalVariable *GV) {
+  VariableHashCalculator GVHC{GV};
+  GVHC.calculateHash();
+  ticketmd::set(GV, GVHC.getHashResult());
```

Comment:
What would happen in this situation... imagine a CU containing something similar to:

    struct S { 
        S() : a{1} {}
        int a;
    }
    S s;
    int t = 1;

The hash assigned by the hash-generation pass for `t` will be  `H(int{1})`; for `s.a` it will be `H(int{0})`. If the optimizer later decides to update `s.a` so that it is initialized to 1, this new code will set its hash to `H(int{1})`. Doest that mean that we will now later try to insert two fragments representing `int{1}`? Does this have any impact?

Reply:
> 
> Doest that mean that we will now later try to insert two fragments representing `int{1}`? 
>
No, only single fragment will be inserted into the database because new fragments is collected in the 'Map' structure (std::map<ticketmd::DigestType, FragmentContentsType>. Since the key is DigestType, there will be only one fragment in the database.

I also tested using the following examples. Only single fragment is stored in the database.

```
target triple = "x86_64-pc-linux-gnu-repo"

define i32 @foo() !repo_ticket !0 {
entry:
  ret i32 1
}

define i32 @bar() !repo_ticket !1 {
entry:
  ret i32 1
}

!repo.tickets = !{!0, !1}

!0 = !TicketNode(name: "foo", digest: [16 x i8] c"N\DB\88\CF\F2a\8D.\8C!yNR[|d", linkage: external, pruned: false)
!1 = !TicketNode(name: "bar", digest: [16 x i8] c"N\DB\88\CF\F2a\8D.\8C!yNR[|d", linkage: external, pruned: false)
```
Compile and dump the fragment:
```
$ llc -filetype=obj repo.ll -o repo.o
$ pstore-dump -all-fragments clang.db
- file      :
      path : clang.db
      size : 4194304
  fragments :
      - digest   : 647c5b524e79218c2e8d61f2cf88db4e
        fragment :
            - type     : text
              contents :
                  align   : 16
                  data    :
                      - 0:      movl    $1, %eax
                      - 5:      retq
                  ifixups : [ ]
                  xfixups : [ ]
...
```

> Does this have any impact?
>
I don't think so.

Reply:
> > Does that mean that we will now later try to `insert` two fragments representing int{1}?

> No, only single fragment will be inserted into the database because new fragments is collected in the 'Map' structure

What if the fragment representing `int{1}` is already in the database?